### PR TITLE
do not add failed relayed tx to invalid mbs

### PR DESCRIPTION
**Pull request by sasurobert**
do not add failed relayed tx to invalid mbs

### CODE REVIEW

File: factory/blockProcessorCreator.go [hunk 1]
```diff
@@ -235,25 +235,26 @@ func (pcf *processComponentsFactory) newShardBlockProcessor(
...
+		ReceiptForwarder:                      receiptTxInterim,
+		BadTxForwarder:                        badTxInterim,
+		ArgsParser:                            argsParser,
+		ScrForwarder:                          scForwarder,
+		RelayedTxEnableEpoch:                  enableEpochs.RelayedTransactionsEnableEpoch,
+		PenalizedTooMuchGasEnableEpoch:        enableEpochs.PenalizedTooMuchGasEnableEpoch,
+		MetaProtectionEnableEpoch:             enableEpochs.MetaProtectionEnableEpoch,
+		EpochNotifier:                         pcf.epochNotifier,
+		RelayedTxV2EnableEpoch:                enableEpochs.RelayedTransactionsV2EnableEpoch,
+		AddFailedRelayedToInvalidDisableEpoch: enableEpochs.AddFailedRelayedTxToInvalidMBsDisableEpoch,
```
Comment (SebastianMarian) [inline factory/blockProcessorCreator.go:257]: AddFailedRelayed**Tx**ToInvalidDisableEpoch

File: process/transaction/shardProcess.go [hunk 2]
```diff
@@ -51,29 +51,33 @@ type txProcessor struct {
 	relayedTxV2EnableEpoch         uint32
 	penalizedTooMuchGasEnableEpoch uint32
 	metaProtectionEnableEpoch      uint32
+
+	flagAddFailedRelayedToInvalid         atomic.Flag
```
Comment (SebastianMarian) [inline process/transaction/shardProcess.go:55]: flagAddFailedRelayed**Tx**ToInvalid**MBs**

File: process/transaction/shardProcess.go [hunk 3]
```diff
@@ -51,29 +51,33 @@ type txProcessor struct {
...
+	addFailedRelayedToInvalidDisableEpoch uint32
```
Comment (SebastianMarian) [inline process/transaction/shardProcess.go:56]: addFailedRelayed**Tx**ToInvalid**MBs**DisableEpoch

File: process/transaction/shardProcess.go [hunk 4]
```diff
@@ -51,29 +51,33 @@ type txProcessor struct {
...
+	EconomicsFee                          process.FeeHandler
+	ReceiptForwarder                      process.IntermediateTransactionHandler
+	BadTxForwarder                        process.IntermediateTransactionHandler
+	ArgsParser                            process.ArgumentsParser
+	ScrForwarder                          process.IntermediateTransactionHandler
+	RelayedTxEnableEpoch                  uint32
+	RelayedTxV2EnableEpoch                uint32
+	PenalizedTooMuchGasEnableEpoch        uint32
+	MetaProtectionEnableEpoch             uint32
+	AddFailedRelayedToInvalidDisableEpoch uint32
```
Comment (SebastianMarian) [inline process/transaction/shardProcess.go:79]: AddFailedRelayed**Tx**ToInvalid**MBs**DisableEpoch

File: process/transaction/shardProcess.go [hunk 5]
```diff
@@ -951,6 +959,9 @@ func (txProc *txProcessor) EpochConfirmed(epoch uint32, _ uint64) {
 
 	txProc.flagMetaProtection.SetValue(epoch >= txProc.metaProtectionEnableEpoch)
 	log.Debug("txProcessor: meta protection", "enabled", txProc.flagMetaProtection.IsSet())
+
+	txProc.flagAddFailedRelayedToInvalid.SetValue(epoch >= txProc.addFailedRelayedToInvalidDisableEpoch)
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/transaction/shardProcess.go]: epoch < txProc.addFailedRelayedToInvalidDisableEpoch

[the remark this answers is not in the document]
Comment (SebastianMarian) [reply process/transaction/shardProcess.go]: It should been more clear if all the flags/arguments/variables from this PR, were been in form like: ...remove....enabled instead ...add....disabled

File: process/transaction/shardProcess.go (hunk 3 above)
Comment (sasurobert) [reply process/transaction/shardProcess.go:56]: That is so long D. Would leave the name as it is.

Comment (iulianpascalau) [reply process/transaction/shardProcess.go:56]: I have already used the suggestion done by Sebi in my branch

Comment (iulianpascalau) [reply process/transaction/shardProcess.go:56]: will even-out this name

**Pull request merged**